R/calculateConfusionMatrix.R

```diff
@@ -54,12 +59,26 @@ calculateConfusionMatrix = function(pred, relative = FALSE, sums = FALSE) {
   n.pred = length(resp)
   truth = getPredictionTruth(pred)
 
+  if (set != "both"){
+      assertClass(pred, classes = "ResamplePrediction")
+      subset.idx = (pred$data$set == set)
+
+      if (!any(subset.idx)){
+          stopf("prediction object contains no observations for set = '%s'", set)
+      }
+      truth = truth[subset.idx]
+      resp = resp[subset.idx]
+  }
+
   cls = union(levels(resp), levels(truth))
   k = length(cls)
   truth = factor(truth, levels = cls)
   resp = factor(resp, levels = cls)
 
   tab = table(truth, resp)
+  if (all(tab == 0)){
+      stop("Confusion matrix is empty!")
+  }
```

Comment:
I don't think we need this check here -- the check above catches the error conditions.

Reply:
Oh, okay. I thought that's what you meant with "make the behavior consistent".

It's removed now.